keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+
+#### Scenarios
+1. A public root associated with a key a Publisher uses to sign artifacts is compromised. The access to root key is compromised, rather than the root key itself being stolen, allowing an attacker to create new intermediate and leaf trusted keys.
+2. An internal root used by an organization is due to expire. A new root is created and rotated before the existing root expires.
+3. An internal root used by an organization is not stored securely and the private key is lost. A new root is created to continue operations.
+4. A cryptographic algorithm is deprecated, or a compliance standard requires a customer to upgrade their key strength. The customer wants to safely rotate a root key without rendering existing signatures invalid and disrupting operations.
+
+### Rescinding Signature Validity
+Artifact Publishers need mechanisms to indicate that a signature they generated is still trustworthy. 
+
+#### Scenarios
+- A Publisher vends a new version of an artifact, and wants to indicate the new version as trusted in addition to older versions already published.
+
+#### Discussion Areas
+- Code signing is a mechanism for Publishers to explicitly indicate trust which is implicitly trusted by Consumer given some conditions are satisfied (based on trust policy, signature being valid and unrevoked)
+    - An explicit allowlist is not required
+    - A denylist is required in addition to indicate which artifacts are no longer trusted
+
+- Signature Allowlist explicitly specifies the list of trusted artifacts.
+    - Pros
+        - Anything not in the list is implicitly untrusted, no separate revocation mechanism is required
+        - If signed allowlist are used, the artifacts themselves may not need to be signed.
+    - Cons
+        - The allowlist needs to be updated for every version of artifact being published
+        - May need to maintain a large allowlist which may be an overhead to distribute
+- Signature Denylist
+    - For Consumers, denylist allows explicitly indicating that an artifact or dependency is untrusted
+    - For Publishers, this allows communicating to Consumers that specific versions of artifact are untrusted.
+
+- Centralized/local public/private lists
+    - The list can be local to repository, requiring update to the list in each repository, and requires keeping track of all repositories where the artifact needs to be published/revoked.
+    - The list can be centralized
+        - Centralized deny list (e.g. CRL maintaned by public CAs) can be used. The endpoint infomation is included in the signature, and signature verification step checks against this list.
+            - Customers can define network topology with restricted network access, where these endpoints may not be accesible for hosts where signature verification occurs.
+            - These endpoints may not be available in air-gapped environments.
```

Comment:
While the endpoints would be unavailable, can we add content the lists can be replicated in the air-gapped environment? We discussed this a bit, where we can't replicate the dns with a different IP as these are not allowed in secure air-gapped environments. 
Just not sure how we allow alternative endpoints without creating man-in-the-middle attacks.

Reply:
Any distribution mechanism would need to be secured to prevent a MitM attack (both for air-gapped environments and for everyone else). This can be done using a delegation from the existing trusted keys, or you could set up another equally trusted mechanism.